site/src/components/decorate-pattern.md

```diff
@@ -0,0 +1,229 @@
+
+Components that implement the decorate pattern expose a `decorate` property which is passed the data join selection used to construct the component's DOM. This allows users of the component to add extra logic to the enter, update and exit selections.
+
+Here are a few examples of how the `decorate` property can be used in practice.
+
+## Rotating axis labels
+
+The d3fc axis re-implements the D3 axis to support the decorate pattern. Each axis tick is composed of a `g` element, which provides the required offset, and has two child elements, a `text` element and a `path` element which renders the tick lines.
+
+In this example the axis labels are rotated by adding a transform to the `text` elements:
+
+```js
+{{{axis-example-code}}}
+```
+
+<style type="text/css">
+  .tick text {
+    font-size: 13px;
+  } 
+</style>
+
+<svg class="axis-container" id="axis-example"></svg>
+
+<script type="text/javascript">
+(function() {
+  var width = $("#axis-example").width();
+
+  var linearScale = d3.scale.linear()
+    .domain([0, 140])
+    .range([0, width])
+    .nice();
+
+  var ordinalScale = d3.scale.ordinal()
+    .domain(['Carrots', 'Bananas', 'Sausages', 'Pickles', 'Aubergines', 'Artichokes', 'Spinach', 'Cucumber'])
+    .rangePoints([0, width], 1);
+
+  {{{axis-example-code}}}
+
+  d3.select('#axis-example')
+    .call(axis);
+}());
+</script>
+
+The data join selection passed to the `decorate` function is the update selection for the `g` elements. The above code accesses the enter selection so that the DOM updates are only applied when ticks are first created. The child `text` element is selected and rotated accordingly.
+
+## Bar colouring
+
+The d3fc series APIs are deliberately very simple, instead you are encouraged to use the decorate pattern in order to configure the series for your needs.
+
+The example below shows how a bar series can be styled to cycle through a color palette:
+
+```js
+{{{bar-example-code}}}
+```
+
+<style type="text/css">
+  .bar path {
+    stroke-width: 0;
+  } 
+</style>
+
+<div id="bar-example" class="chart" style="height: 200px"> </div>
+
+<script type="text/javascript">
+(function() {
+  var f = createFixture('#bar-example', null, 200, 10, function() { return true; });
+  var container = f.container, data = f.data,
+    xScale = f.xScale, yScale = f.yScale;
+
+  {{{bar-example-code}}}
+
+  container.append('g')
+      .datum(data)
+      .call(bar);
+}());
+</script>
+
+Again, the enter selection is used and the required child element, in this case a `path` is selected. D3 selections allow you to specify values as functions of the bound data and index, here the index is used to cycle through a color scale.
```

Comment:
> D3 selections allow you to

I'm not sure if that's quite right but I'm not sure of a better way of wording it.